src/procedures/__tests__/SetController.ts

```diff
@@ -0,0 +1,181 @@
+import { ImportMock, MockManager } from 'ts-mock-imports';
+import { SinonStub, stub, spy, restore } from 'sinon';
+import * as contractWrappersModule from '@polymathnetwork/contract-wrappers';
+import * as contextModule from '../../Context';
+import * as wrappersModule from '../../PolymathBase';
+import * as tokenFactoryModule from '../../testUtils/MockedTokenFactoryObject';
+import { SetController } from '../../procedures/SetController';
+import { Procedure } from '~/procedures/Procedure';
+import {
+  ProcedureType,
+  PolyTransactionTag,
+  SetControllerProcedureArgs,
+  ErrorCode,
+} from '~/types';
+import { PolymathError } from '~/PolymathError';
+import * as utilsModule from '../../utils';
+import { Wallet } from '../../Wallet';
+
+const params1 = {
+  symbol: 'TEST1',
+  controller: '0x3333333333333333333333333333333333333333',
+};
+
+describe('SetController', () => {
+  let target: SetController;
+  let contextMock: MockManager<contextModule.Context>;
+  let wrappersMock: MockManager<wrappersModule.PolymathBase>;
+  let tokenFactoryMock: MockManager<
+    tokenFactoryModule.MockedTokenFactoryObject
+  >;
+  let securityTokenMock: MockManager<
+    contractWrappersModule.SecurityToken_3_0_0
+  >;
+
+  let tokenFactoryMockStub: SinonStub<any, any>;
+
+  beforeEach(() => {
+    // Mock the context, wrappers, and tokenFactory to test AssignSecurityRole
+    contextMock = ImportMock.mockClass(contextModule, 'Context');
+    wrappersMock = ImportMock.mockClass(wrappersModule, 'PolymathBase');
+    tokenFactoryMock = ImportMock.mockClass(
+      tokenFactoryModule,
+      'MockedTokenFactoryObject'
+    );
+    securityTokenMock = ImportMock.mockClass(
+      contractWrappersModule,
+      'SecurityToken_3_0_0'
+    );
+
+    tokenFactoryMockStub = tokenFactoryMock.mock(
+      'getSecurityTokenInstanceFromTicker',
+      securityTokenMock.getMockInstance()
+    );
+
+    contextMock.set('contractWrappers', wrappersMock.getMockInstance());
+    wrappersMock.set('tokenFactory', tokenFactoryMock.getMockInstance());
+  });
+
+  afterEach(() => {
+    restore();
+  });
+
+  describe('Types', () => {
+    test('should extend procedure and have SetController type', async () => {
+      target = new SetController(
+        {
+          symbol: params1.symbol,
+          controller: params1.controller,
+        },
+        contextMock.getMockInstance()
+      );
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe('SetController');
+    });
+  });
+
+  describe('SetController', () => {
+    test('should throw if there is no valid security token being provided', async () => {
+      // Instantiate SetController with incorrect security symbol
+      target = new SetController(
+        {
+          symbol: params1.symbol,
+          controller: params1.controller,
```

Comment:
Instantiate please with just params (I’ve gotten recent review from Jere about this type of issue)

Reply:
Simplify the code just passing params object in